### DISCUSSION TITLE
Ruby Client Support

### DISCUSSION THREAD

**Discussion by dominicfollett** [apache/pulsar#17781] [Q&A]
Is there anything on the roadmap for providing a Pulsar Ruby Client?

**Comment by michaeljmarshall** [comment]
I don't know of any active work being done on a Pulsar Ruby Client. I know almost nothing about Ruby, but is it possible to use the Pulsar C++ Client in your Ruby code? Our Python client is currently a wrapper around the C++ client.

For reference, we enumerate official and third party clients here: https://pulsar.apache.org/docs/next/client-libraries.

**Comment by tisonkun** [comment]
There is a http://github.com/apache/pulsar-client-ruby repo but it's halted. There were several efforts trying to wrap the C++ client to a Ruby gem and you may find some released gems at https://rubygems.org/gems/pulsar-client. However, the development stalled for a long time. So using it is at your risk.

**Reply by tisonkun** [reply]
One alternative is using the [REST](https://pulsar.apache.org/docs/next/client-libraries-rest/) or [WebSocket](https://pulsar.apache.org/docs/next/client-libraries-websocket/) interface which is language-agnostic. IIRC @pingcap builds their showcase that combines https://ossinsight.io/ with Pulsar in Ruby by this approach.
